hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -71,6 +78,22 @@ func RecordNodePoolAvailableReplicas(nodePool *hyperv1.NodePool) {
 	nodePoolAvailableReplicas.With(labels(nodePool)).Set(float64(nodePool.Status.Replicas))
 }
 
+func RecordNodePoolInfo(nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster) {
+	labels := labels(nodePool)
+
+	// Fill the extra info there
+	npPlatform := nodePool.Spec.Platform
+
+	labels["_id"] = hcluster.Spec.ClusterID
+	if npPlatform.Type == hyperv1.AWSPlatform && npPlatform.AWS != nil {
+		labels["ec2_instance_type"] = npPlatform.AWS.InstanceType
+	} else {
+		labels["ec2_instance_type"] = ""
+	}
+
+	nodePoolInfo.With(labels).Set(1)
```

Comment:
This will generate a new metric (gauge) for every new value of `ec2_instance_type` label, i.e. whenever the nodePool spec.InstanceType is updated. So we will be getting multiple `hypershift_nodepools_info` metrics for the same nodePool.

We need to cleanup/delete old metrics before generating new ones.

Reply:
Yeah I am designing a new custom collector as a workaround to that problem.

Reply:
@muraee, here is the PR using collectors for the metrics: https://github.com/openshift/hypershift/pull/2671.
I will migrate the code to this collector based architecture.